provider/functionality_resource.go

```diff
@@ -225,6 +230,11 @@ func deleteResourceFireHydrantFunctionality(ctx context.Context, d *schema.Resou
 	functionalityID := d.Id()
 	err := firehydrantAPIClient.DeleteFunctionality(ctx, functionalityID)
 	if err != nil {
+		_, isNotFoundError := err.(firehydrant.NotFound)
```

Comment:
Out of curiosity: is there a reason not to use the `errors.Is(err, firehydrant.NotFound)` package/function for doing this rather than type assertion on an interface?

Reply:
basically just that i couldn't get it to work with the way the NotFound error was defined without doing this:
```
errors.Is(err, firehydrant.NotFound("resource not found"))
```
which felt really weird to me. i think i'm either missing something (most likely) or the NotFound error is defined in a weird way that could be improved. i didn't want to touch too much in the client since i think we plan to replace/rewrite it so i switched it to that type assertion.